### PR TITLE
feat(UI): search regex or exact match

### LINKED ISSUE #66: Add additional search methods: exact, regex
### Warframe Drops Site Feature Request
---

**Summary (short):**
Provide the option to search by exact string

Provide the option to search using regex

---
**Detailed requirements of the feature**

_As a user, I should be able to indicate that I wish to search for a drop using an exact search string._
_As a user, those search results should match the search string **entirely**._

_As a user, I should be able to search for a drop using a regex._
_As a user, those search results should match the regex._

---
**Mockups:**

Paint, photoshop mockups, sketchup, whatever you're envisioning.

A tickbox to indicate whether or not "Exact match" is desired
If this tickbox is selected, then searching for "Bite" should match "Bite" *only*, and not Frostbite. Searching for "Rush" should match "Rush" only, and not "Blood Rush", or "Gladiator Rush"

A tickbox to indicate whether or not "regex match" is desired
If this tickbox is selected, then searching for "`^bite$`" should match "Bite" *only*, and not Frostbite. Similarly, searching for "`^(gladiator )?rush$`" would match "Rush", "Gladiator Rush" but not "Blood Rush"

---

**Additional context**
Add any other context about the request here.

### PR DESCRIPTION
**Pull request by AyAyEm**
### Warframe Drop Data Site Pull Request

---

**What I did:**
Added regex, exact and default match options as a dropdown selection.

---
**Why I did it:**
To search with more precision for specific drops, locations etc..

---
**Fixes issue (include link):**
closes #66 

---
**Mockups, screenshots, evidence:**

**Before:**
image
image

**After:**
image
image
image
image

---

**Was this an issue fix or a suggestion fulfillment?**
Suggestion fulfillment

### CODE REVIEW

**Comment by AyAyEm** [comment]
Sorry for the late changes, when i woke up, i remembered that it would be nice to have some of the things that i pushed today. It should be good to review now.

File: site/index.html [hunk 1]
```diff
@@ -50,6 +50,11 @@ <h5>Warframe Drop Data</h5>
 
             <div>
                 <input class="searchfield" placeholder="Search here..." id="search-field" type="text" autofocus="autofocus" aria-label="Search here..." />
+                <select id="search-match" aria-label="Search Matching">
```
Comment (TobiTenno) [inline site/index.html:53]: This block seems a bit too indented

File: site/js/index.js [hunk 2]
```diff
@@ -125,20 +131,35 @@ function search(searchValue) {
...
 
-    window.location.hash = `/search/${encodeURIComponent(searchValue)}/${type}`
+    window.location.hash = `/search/${encodeURIComponent(searchValue)}/${type}/${match}`
+
+    const lowerSearchValue = searchValue.toLowerCase()
+    const regex = match === 'regex' ? new RegExp(searchValue, 'i') : null
+    function matchPredicate(value) {
+        const lowerValue = value.toLowerCase()
+
+        if (match === 'exact') {
```
Comment (TobiTenno) [inline site/js/index.js:146]: Switch instead of if/else chain

**Review by TobiTenno — CHANGES_REQUESTED** [review]

File: site/index.html (hunk 1 above)
Comment (AyAyEm) [reply site/index.html:53]: Seems fine to me, but you spotted a subtle bug, i forgot to include index.html in a commit and search-match should be match-search

**Review by TobiTenno — APPROVED** [review]

**Pull request merged**